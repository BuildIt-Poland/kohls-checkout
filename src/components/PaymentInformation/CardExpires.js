import styled from 'styled-components';

import { default as DefaultText } from '../Text';

const CardExpires = styled(DefaultText)`
  grid-column-start: 2;
  grid-row-start: 2;
`;

export default CardExpires;
