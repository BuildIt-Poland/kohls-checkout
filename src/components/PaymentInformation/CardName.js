import React from 'react';
import styled from 'styled-components';

import SectionHeader from '../SectionHeader';
import { AMEX, MASTERCARD, VISA } from '../../constants/paymentCardsTypes';

const Text = styled(SectionHeader)`
  grid-column-start: 2;
  grid-row-start: 2;
`;

function CardName({ cardNumber }) {
  return (
    <>
      {(() => {
        switch (true) {
          case VISA().test(cardNumber):
            return <Text>Visa</Text>;
          case MASTERCARD().test(cardNumber):
            return <Text>Master Card</Text>;
          case AMEX().test(cardNumber):
            return <Text>American Express</Text>;
          default:
            return null;
        }
      })()}
    </>
  );
}

export default CardName;
