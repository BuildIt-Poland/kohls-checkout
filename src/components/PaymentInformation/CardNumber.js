import styled from 'styled-components';

import { default as DefaultText } from '../Text';

const CardNumber = styled(DefaultText)`
  grid-column-start: 2;
  grid-row-start: 3;
`;

export default CardNumber;
