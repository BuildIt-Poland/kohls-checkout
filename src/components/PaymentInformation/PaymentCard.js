import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import { default as DefaultPaymentCard } from '../PaymentCard';

const Wrapper = styled.div`
  grid-column-start: 1;
  grid-row-start: 1;
`;

function PaymentCard({ cardNumber }) {
  return (
    <Wrapper>
      <DefaultPaymentCard cardNumber={cardNumber} />
    </Wrapper>
  );
}

PaymentCard.propTypes = {
  cardNumber: PropTypes.string.isRequired
};

export default PaymentCard;
