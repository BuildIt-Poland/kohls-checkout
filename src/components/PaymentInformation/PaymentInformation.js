import React from 'react';
import { Link } from 'react-router-dom';

import { CHECKOUT_PAYMENT_PATH } from '../../constants/routes';
import TextLink from '../TextLink/TextLink';
import SectionHeader from '../SectionHeader';

function PaymentInformation() {
  const textLink = (
    <TextLink as={Link} to={CHECKOUT_PAYMENT_PATH}>
      Edit
    </TextLink>
  );

  return (
    <section>
      <SectionHeader actionElement={textLink}>Payment Information</SectionHeader>
    </section>
  );
}

export default PaymentInformation;
