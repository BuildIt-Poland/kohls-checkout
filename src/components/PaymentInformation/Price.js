import styled from 'styled-components';

import { default as DefaultText } from '../Text';

const Price = styled(DefaultText)`
  grid-column-start: 3;
  grid-row-start: 2;
`;

export default Price;
