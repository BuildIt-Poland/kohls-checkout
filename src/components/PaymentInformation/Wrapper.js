import styled from 'styled-components';

import media from '../../styles/media';

const Wrapper = styled.div`
  display: grid;
  grid-template-columns: auto 1fr;
  grid-gap: 20px;

  ${media.phone`
    display: block;
  `};
`;

export default Wrapper;
