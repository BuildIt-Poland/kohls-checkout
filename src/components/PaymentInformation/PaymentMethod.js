import React from 'react';
import PropTypes from 'prop-types';

import Wrapper from './Wrapper';
import PaymentCard from './PaymentCard';
import CardName from './CardName';
import Text from '../Text';
import Price from './Price';

function PaymentMethod({ cardNumber, expiries, price }) {
  const maskCardNumber = cardNumber => cardNumber.replace(/^(\d{4}){3}(?=\d{4})/gi, 'XXXX-XXXX-XXXX-');

  return (
    <Wrapper>
      <PaymentCard cardNumber={cardNumber} />
      <div>
        <CardName cardNumber={cardNumber} />
        <Text as="span">{maskCardNumber(cardNumber)}</Text>
        <Price as="span" bold>
          ${price}
        </Price>
        <Text>Expires {expiries}</Text>
      </div>
    </Wrapper>
  );
}

PaymentMethod.propTypes = {
  cardNumber: PropTypes.string.isRequired,
  expiries: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired
};

export default PaymentMethod;
