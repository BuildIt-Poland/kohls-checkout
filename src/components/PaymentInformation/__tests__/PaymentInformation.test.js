import React from 'react';
import { render } from 'react-testing-library';

import PaymentInformation from '../PaymentInformation';

jest.mock('react-router-dom', () => ({
  Link: 'MockedLink'
}));

describe('COMPONENT - PaymentInformation', () => {
  it('renders PaymentInformation component', () => {
    const { container } = render(<PaymentInformation />);

    expect(container.querySelector('h2')).toHaveTextContent('Payment Information');
  });
});
