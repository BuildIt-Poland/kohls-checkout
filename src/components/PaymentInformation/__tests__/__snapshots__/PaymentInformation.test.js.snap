// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - PaymentInformation renders PaymentInformation component 1`] = `
.c1 {
  color: #00a09f;
  font-weight: 700;
  font-size: 1.8rem;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-transform: none;
}

.c0 {
  font-size: 1.2rem;
  text-align: left;
  margin-top: 10px;
  margin-bottom: 10px;
  text-transform: uppercase;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<section>
  <h3
    className="c0"
  >
    Payment Information
    <MockedLink
      className="c1"
      to="/checkout/payment"
    >
      Edit
    </MockedLink>
  </h3>
</section>
`;
